bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTreeViewer.java

```diff
@@ -1817,18 +1854,23 @@ protected Widget internalGetWidgetToSelect(Object elementOrTreePath) {
 	}
 
 	/**
-	 * Recursively expands the subtree rooted at the given widget to the given
-	 * level.
+	 * Recursively, conditionally expands the subtree rooted at the given widget to
+	 * the given level. Takes the {@code shouldChildrenExpand} predicate that
+	 * defines for a given Widget if it shall be expanded.
 	 * <p>
 	 * Note that the default implementation of this method does not call
 	 * <code>setRedraw</code>.
 	 * </p>
 	 *
-	 * @param widget the widget
-	 * @param level  non-negative level, or <code>ALL_LEVELS</code> to collapse all
-	 *               levels of the tree
+	 * @param widget               the widget
+	 * @param level                non-negative level, or <code>ALL_LEVELS</code> to
+	 *                             collapse all levels of the tree
```

Comment:
This is about expand and not collapse, is it? I know the documentation was like this before, but probably it was incorrect?
```suggestion
	 *                             expand all levels of the tree
```

Reply:
The expansion or collapsing of the widget depends on the third parameter to this method;
```java
if (level == ALL_LEVELS || level > 1) {
Item[] children = getChildren(widget);
if (children != null) {
	int newLevel = (level == ALL_LEVELS ? ALL_LEVELS
			: level - 1);
	for (Item element : children) {
		if (shouldChildrenExpand.apply(widget).booleanValue()) {
			internalConditionalExpandToLevel(element, newLevel, shouldChildrenExpand);
		}
	}
}
}
```
As such, neither words are really well-fit to fully describe the behaviour. What about something like "non-negative level, or {@code ALL_LEVELS} to which paths of a tree should either be collapsed or expanded"?

Reply:
From my understanding, the description is fine as already proposed. My concern was only regarding "collapsed" as the method only seems to expand. Is there really functionality to collapse inside this method?

Reply:
I have double-checked, you are indeed right, there is no code to collapse widgets in the method, hence `expand` is the right terminology